class-wc-calypso-bridge-dotcom-features.php

```diff
@@ -340,4 +360,62 @@ public static function is_business_plan() {
 
 		return self::$is_business_plan;
 	}
+
+	public static function is_trial_plan() {
+		if ( is_null( self::$is_trial_plan ) ) {
+			self::$is_trial_plan = self::is_ecommerce_trial_plan()
+				// Business trial plans
+				|| self::has_plan( 'wp-bundle-hosting-trial', true )
+				|| self::has_plan( 'wp-bundle-migration-trial', true );
+		}
+
+		return self::$is_trial_plan;
+	}
+
+	/**
+	 * Check if the site has a specific plan.
+	 *
+	 * @param string $plan The plan to check for.
+	 * @param bool $exact_one_plan If true, the site must have exactly one plan.
+	 *
+	 * @return bool True if the site has the plan, false otherwise.
+	 */
+	/**
+	 * Check if the site has a specific plan.
+	 *
+	 * This method checks if the current site has a particular plan based on its purchases.
+	 * It can optionally check if the site has exactly one plan.
+	 *
+	 * @param string $plan           The plan slug to check for.
+	 * @param bool   $exact_one_plan If true, the site must have exactly one plan.
+	 *
+	 * @return bool True if the site has the specified plan, false otherwise.
+	 */
+	private static function has_plan( $plan, $exact_one_plan = false ) {
```

Comment:
Nit: Would it be better if we support passing array for `$plan`? I'm unsure how performant is this check, and we have a prior art in checking for [multiple plan string](https://github.com/Automattic/wc-calypso-bridge/pull/1507/files#diff-c9506099f3f037eaa3fc4b48305a4d641f4204266a5494640c4876e942ea711aR329).

Also, I think it's better to have `$exact_one_plan` check to `true` since it's how it was done before in similar checks

Reply:
Sounds good! Updated it here 0ec8e66b3364aa3318021429b07610f50b220f24